src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -41,7 +41,7 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
 
             // Specialization can change the CompatMode setting, so this must run later than
             // the PlaceholderSpecializationMiddleware
-            builder.UseWhen(context => httpBodyControlOptions.CurrentValue.AllowSynchronousIO, config =>
+            builder.UseWhen(context => httpBodyControlOptions.CurrentValue.AllowSynchronousIO || context.Request.IsAdminDownloadRequest(), config =>
```

Comment:
I thought all of the code in the download api was using async methods all the way through. Or is it just the case that file IO in .net is always synchronous at some point?

Reply:
Even if download APIs are async, IISServer seems to do a "Write" (and not WriteAsync) internally at some point to start writing to response stream which is causing an exception. This fails the downloadAPI silently and causes an empty zip to be downloaded.

Reply:
Thanks for the details. I guess I only ever tested it without IIS in the picture, but that's good to know.